spec/models/user_spec.rb

```diff
@@ -82,4 +82,43 @@
       end
     end
   end
+
+  describe '#create' do
+    context 'can save user' do
+      it 'is valid with a name, email, password, password_confirmation and location' do
+        user = build(:user)
```

Comment:
For this you should extract a let:
```
let(:user_params) { {} }
let(:user) { build(:user, user_params) }
```
and then only when needed you can use:
```
let(:user_params) { { location: nil } }
```

But in the end you should be using `User.create` to check if things work properly for creating the model and as @ag4ta  suggested `validations` should be tested separately, also no need to check `#edit` - unless you have actual code that's run only there.